routes/login.js

```diff
@@ -0,0 +1,55 @@
+let newToken = require('../utils/newToken');
+let bcrypt = require('bcrypt');
+
+let mongoSchema = require('../config/mongoSchema');
+let User = mongoSchema.User;
+
+let express = require('express');
+let router = express.Router();
+
+router.post('/', function(req, res, next) {
+  console.log('req.body: ', req.body);
+
+  let {username, password} = req.body;
+  if (!username || !password) {
+    return res.status(400).json({ // bad request
+      message: "login failed, please check your username/password"
+    })
+  }
+
+  User.findOne({username}, (err, user) => {
+    console.log('fetched user: ', user);
+    if (err){
+      return res.status(500).json({
+        message: "server error, please retry later"
+      });
+    }
+
+    if (!user) {
+      console.log("username not exists");
+      return res.status(400).json({
+        message: "login failed, please check your username/password"
+      });
+    }
+
+    bcrypt.compare(password, user.password, (err, isMatched) => {
+      if (err){
+        return res.status(500).json({
+          message: "server error, please retry later"
+        });
+      }
+
+      if (isMatched){
+        let token = newToken(username);
+        res.status(200).json({token, username})
+      } else {
+        console.log("wrong password");
+        res.status(401).json({
```

Comment:
I'm curious why you return 400 for https://github.com/Wiredcraft/test-fullstack/pull/27/files#diff-3fb8e84150106440c8ef0ddca2ef14d5R30, and 401 for mismatch of id/pwd .
Do you see any issue if the response code are not the same for these two cases?

Reply:
Yes, you're right, server side should return 200(login succeeded), 400(login failed, or 401), 500(server error).